code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -15,239 +152,216 @@
 	materials = list(MAT_METAL=500)
 	resistance_flags = FIRE_PROOF
 	origin_tech = "combat=1;plasmatech=2;engineering=2"
-	var/status = FALSE
-	var/lit = FALSE	//on or off
-	var/operating = FALSE//cooldown
-	var/obj/item/weldingtool/weldtool = null
-	var/obj/item/assembly/igniter/igniter = null
-	var/obj/item/tank/plasma/ptank = null
-	var/warned_admins = FALSE //for the message_admins() when lit
-	//variables for prebuilt flamethrowers
+	flame_intensity = 1
+	consumption_rate = 25
+	tank_exposure = 15
+	lit = FALSE	
+	var/obj/item/weldingtool/weldtool = null //so you can recover the welding tool you used
 	var/create_full = FALSE
 	var/create_with_tank = FALSE
-	var/igniter_type = /obj/item/assembly/igniter
 
-
-/obj/item/flamethrower/Destroy()
+/obj/item/flamethrower/basic/Destroy()
 	QDEL_NULL(weldtool)
-	QDEL_NULL(igniter)
 	QDEL_NULL(ptank)
```

Comment:
Why have this on the basic and advanced level and not on the flamethrower level?